### DISCUSSION TITLE
How to use with NextJS 14

### DISCUSSION THREAD

**Discussion by stychu** [LegendApp/legend-state#238] [Q&A]
I have a nextjs 14.0.4 project and wanted to try legend state. But Im not able to get the rerender of the changed value.
I've created a teststore.ts file with content 
```ts
import { observable } from '@legendapp/state';

export const schema = observable('abrakadabra');
```
and then displayed this in a client component and a server component.

Added a button with onclick to change the value

```ts
  <button onClick={() => schema.set(Math.floor(Math.random() * 1000))}>
        Click
      </button>
```
added enableReactTracking({ auto: true });
 in bunch of places but id doesnt trigger rerenders.  When I do soft navigation the value for the client component changes but server component does not.

**Comment by jmeistrich** [comment]
It doesn't work well across server <-> client. Having global server state wouldn't really make sense because it would be shared across all users, and there's not a great generic way that I'm aware of to synchronize state between client and server.

But if you do think it should work and we could find a way to solve it, can you share more of your code?
